### ISSUE TITLE
Remove dependency of make version_test/unit-test/bash_tests on bash

### ISSUE THREAD

**Issue by clanmills** [Exiv2/exiv2#1274]
After #1257 is merged, we can now move rapidly to remove the last vestiges of bash in the test suite.

Currently the commands in the build directory use bash to run different parts of the test suite. 

| bash | cmd.exe |
|:--    |:--           |
| $ make bash_tests | cmake --build . --config Release --target bash_tests |
| $ make unit_test | cmake --build . --config Release --target unit_test |
| $ make version_test | cmake --build . --config Release --target bash_tests |57,

Remove the dependency on bash.  In #1257, you are running the new bash_tests from python.  Let's give similar treatment to unit_test and version_test.

Please prepare the PR for v0.27-maintenance.  We'll have to update README.md (and possibly README-CONAN.md) to match the code.  I'll give you those changes as part of the PR review.

There is no plan yet to adopt #1257 and other python test enhancements into 'master'.

We've had an offer from @kmilos to help test this on MinGW/msys2.  Thank You, @kmilos.  I'll get GitHub to invite you to join Team Exiv2 and I will add you to the assignees for this issue and subsequent PR.  I won't dump work on @kmilos unless he volunteers for other issues.

**Comment by LeoHsiao1** [comment]
Sorry, I still don't know about unit_test and version_test.
Are we going to rewrite this part of the content in the Makefile?
```py
version_test \
unit_test    :
	@echo
	@echo ---- Running $@ ----
	@echo
	-@./$@.sh
```

Or rewrite the test cases named version_test.py and unit_test.py?

**Comment by clanmills** [comment]
I think running unit_test from python is a couple of lines on python and version_test is one line of python!

**make unit_test**
```
$ cmake .. -DEXIV2_BUILD_UNIT_TESTS=1
$ make unit_tests        # create the excitable bin/unit_tests
$ make unit_test          # runs the executable bin/unit_tests  
```
Today, it runs some bash code:
```bash
1081 rmills@rmillsm1:~/gnu/github/exiv2/0.27-maintenance/build $ cat ../test/unit_test.sh 
#!/usr/bin/env bash
# driver for $ make unit_test

( source ./functions.source ; runTest unit_tests --gtest_color=no 2>&1 | grep -v "Warning: Unsupported .... format" )

# That's all Folks!
##
```

Can you get `$ make unit_test` run something like `python3 runner something.py`  And something.py should be run LAST by `$ python3 runner.py` 

**make version_test**

Every build of exiv2 command-line program supports options `--verbose --version`.  There's a little bash magic to filter/reduce the output from the command in the test suite.  I think you only need to call:

`system_tests.BT.verbose_version(True)`.

You'll do this easily when you understand what I'd like you to do!

Remember the aim.  We want bash out of the code base.  Gone.  Unix users will run the test suite with command such as:

```bash
$ cd build
$ make tests
$ make version_test  etc....
```

msvc developers will use:
```bash
$ cd build
$ cmake --build . --target tests
$ cmake --build . --target version_test etc...
```
Both those command execute `python3 runner.py [magic.py]` to do the work.

**Comment by LeoHsiao1** [comment]
Oh oh, I won't have time to finish it this weekend.

**Comment by clanmills** [comment]
That's fine.  I'm going to release Exiv2 v0.27.4 RC1 this week.  We are two weeks ahead of schedule.

Code freeze/RC2 is scheduled for 2021-04-30.  However it would be very helpful if you could complete this in March and we arrive at Code Freeze in Early April.

**Comment by clanmills** [comment]
Closed by #1495
